cdap-api/src/main/java/co/cask/cdap/api/lineage/Operations.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Graph representing set of {@link Read} operations from the data sources,
+ * set of {@link Transform} operations, and set of {@link Write} operations to the data sinks.
+ */
+public class Operations {
+  private final Set<Read> reads;
+  private final Set<Transform> transforms;
+  private final Set<Write> writes;
+
+  /**
+   * Create Operations instance.
```

Comment:
the longer I think about this, the less I feel that this representation is unambiguous. Consider this example:

```
read: dataIn -> (name, ssn)
write: (name, ssn) -> originalOut
transform: (name) -> toUpperCase, "convert all letters to upper case" -> (name)
write: (name, ssn) -> derivedOut
```
Basically a pipeline that forks, and in one branch it transforms a field without renaming it. 

From looking at these operations, we cannot tell whether either of the outputs has the original or the upper case version of the name. 

We could say this is invalid and reject it - but it is a perfectly legal pipeline.

Reply:
The ambiguity is because of the same field `name` is used at multiple places. I think we discussed about it and we said that nodes in the graph need to have unique names. For example, we have two file sources both emit offset and body fields, however to uniquely identify the body or offset field we will have to prefix it with stage name for example - <stage.name>.body and <stage.name>.offset